### DISCUSSION TITLE
Support for OpenAI compatible api provider

### DISCUSSION THREAD

**Discussion by Learner00100** [Taewan-P/gpt_mobile#62] [Ideas]
Hi,
Thanks for this wonderful project.
recently you implement custom endpoint, if you able to implement custom name also we can make work all existing OpenAI compatible provider like perplexity , mistral, etc.
make sure implementation is on latest openai compatible Api to works smooth.

if this is achieved, try to implement adding Mutiple openai api chat to enabled in app.
so that we can have many custom llm of openai compatibl

**Comment by Taewan-P** [comment]
This can not be considered until https://github.com/Taewan-P/gpt_mobile/issues/41#issuecomment-2337878814 gets solved due to same reasons.

**Comment by 11111000000** [comment]
Any work on that?

**Comment by Taewan-P** [comment]
Not yet. I was pretty busy these days..I have to finish Ollama support first

**Comment by nitanmarcel** [comment]
@Taewan-P I've just tested openrouter's endpoint and it seems to work fine without any modifications to the openai implementation 
![Screenshot_20241013-160315](https://github.com/user-attachments/assets/06272bec-7116-4f60-a5ae-db2b4a0aaf43)
![Screenshot_20241013-160225](https://github.com/user-attachments/assets/cbfeceda-1432-476b-a740-34ccf7dd6538)

**Reply by Taewan-P** [reply]
@Learner00100 @11111000000 @nitanmarcel Yes for that it would work since custom endpoint and custom models are now supported. It's kinda like a workaround...That's why I implemented this first.

For custom name & multiple same providers; This is currently impossible to implement due to the codebase 😅 Needs whole refactoring

**Comment by Taewan-P** [comment]
Hello, this feature is now implemented in [v0.7.0](https://github.com/Taewan-P/gpt_mobile/releases). Thanks!
cc @Learner00100 @11111000000 @nitanmarcel
